kornia/geometry/calibration/undistort.py

```diff
@@ -105,7 +105,7 @@ def undistort_image(image: torch.Tensor, K: torch.Tensor, dist: torch.Tensor) ->
     distortion models are considered in this function.
 
     Args:
-        image: Input image with shape :math:`(*, C, H, W)`.
+        image: Input image with shape :math:`(*, N, C, H, W)`.
```

Comment:
don't get why we need to add an extra dimension `N` here ?

Reply:
My bad, I now realise that this `N` should not be here. I got confused due to the following (pre-PR) behaviour:
```python
from kornia.geometry.calibration import undistort_image
import torch

# Raises ValueError
image = torch.rand(3, 5, 5)
K = torch.rand(3, 3)
dist = torch.rand(4)
undistort_image(image, K, dist)

# Works, therefore I thought `N` should be included
image = torch.rand(1, 3, 5, 5)
K = torch.rand(3, 3)
dist = torch.rand(4)
undistort_image(image, K, dist)

# Raises RuntimeError, therefore I realise the previous case just got lucky
image = torch.rand(2, 3, 5, 5)
K = torch.rand(3, 3)
dist = torch.rand(4)
undistort_image(image, K, dist)

# Works, therefore I realise that `N` should not be included
image = torch.rand(2, 3, 5, 5)
K = torch.rand(2, 3, 3)
dist = torch.rand(2, 4)
undistort_image(image, K, dist)
```
It looks like [this](https://github.com/kornia/kornia/blob/80c6054c1fb8e687ef7cd6167b294e474e99ccc7/test/geometry/calibration/test_undistort.py#L207) existing test is testing the wrong shape. I guess that test should be kept so that future changes don't break existing code, however, I have added an additional test called `test_shape_minimum_dims` that tests the correct shape. Should I add a comment in `test_shape` explaining the incorrect shape?

Reply:
ideally we should have exception tests like in here https://github.com/kornia/kornia/blob/master/test/color/test_gray.py#L21

Reply:
I've now added exception tests.